### ISSUE TITLE
Cannot install any Steam app

### ISSUE THREAD

**Issue by Kreyren** [PhoenicisOrg/scripts#959]
None of steam games are successfully installed on phoenicis

Actuall result is that phoenicis installs steam and halts on `Downloading runtime json`

```
[INFO ] org.phoenicis.tools.archive.Tar (l.172) - Creating output file /home/kreyren/.Phoenicis/engines/wine/upstream-linux-x86/4.0/./bin/winepath (493).
[INFO ] org.phoenicis.tools.archive.Tar (l.172) - Creating output file /home/kreyren/.Phoenicis/engines/wine/upstream-linux-x86/4.0/./bin/winedbg (493).
[INFO ] org.phoenicis.tools.archive.Tar (l.172) - Creating output file /home/kreyren/.Phoenicis/engines/wine/upstream-linux-x86/4.0/./bin/msiexec (493).
[INFO ] org.phoenicis.tools.archive.Tar (l.172) - Creating output file /home/kreyren/.Phoenicis/engines/wine/upstream-linux-x86/4.0/./bin/winefile (493).
[INFO ] org.phoenicis.tools.archive.Tar (l.172) - Creating output file /home/kreyren/.Phoenicis/engines/wine/upstream-linux-x86/4.0/./bin/winemaker (493).
Warning: Nashorn engine is planned to be removed from a future JDK release
[WARN ] org.phoenicis.javafx.views.mainwindow.installations.dto.InstallationDTO (l.56) - Installation ID (Mafia II_1559755098177) contains invalid characters, will remove them.
Warning: Nashorn engine is planned to be removed from a future JDK release
Warning: Nashorn engine is planned to be removed from a future JDK release
Warning: Nashorn engine is planned to be removed from a future JDK release
Warning: Nashorn engine is planned to be removed from a future JDK release
Warning: Nashorn engine is planned to be removed from a future JDK release
[WARN ] org.phoenicis.javafx.views.mainwindow.installations.dto.InstallationDTO (l.56) - Installation ID (Mafia II_1559755330515) contains invalid characters, will remove them.
Warning: Nashorn engine is planned to be removed from a future JDK release
[WARN ] org.phoenicis.javafx.views.mainwindow.installations.dto.InstallationDTO (l.56) - Installation ID (Mafia II_1559755338442) contains invalid characters, will remove them.
```

To reproduce install Mafia II on https://github.com/PhoenicisOrg/phoenicis/commit/15bc8a0f15afca5556cb332ee64b718ddd39155f

**Comment by ImperatorS79** [comment]
Is there a runtime folder in .Phoenicis/engine/wine ? If so, that's just that the latest message displayed is "Download runtime .json", but it does not hang there.

**Comment by Kreyren** [comment]
> Is there a runtime folder in .Phoenicis/engine/wine ? If so, that's just that the latest message displayed is "Download runtime .json", but it does not hang there.

`~/.Phoenicis/engine/wine/runtime.json` has http://ix.io/1L0L, but game is not started and won't start the download.

**Comment by Kreyren** [comment]
Same issue on MinionMasters

![image](https://user-images.githubusercontent.com/11302521/58979464-c1a0e480-8793-11e9-84ab-d291625354f4.png)

**Comment by Kreyren** [comment]
Updated OP

**Comment by plata** [comment]
The log in the pull request description does not contain any error. Also, if the download halts during the runtime download, this should not be related to Steam at all but rather affect every app.

**Comment by Kreyren** [comment]
> The log in the pull request description does not contain any error. Also, if the download halts during the runtime download, this should not be related to Steam at all but rather affect every app.

Because there is none error and i'm not familiar with that code.. feel free to reproduce by installing any steam app that is using quick steam script on phoenicis

**Comment by Zemogiter** [comment]
I think I have a similiar error but when I tried sims 2 origin install I get this shortly after downloading runtime json:
```
Unable to extract the file
	at org.phoenicis.tools.archive.Tar.uncompress(Tar.java:199)
	at org.phoenicis.tools.archive.Tar.uncompressTarGzFile(Tar.java:70)
	at org.phoenicis.tools.archive.Extractor.uncompress(Extractor.java:63)
	at org.phoenicis.tools.archive.Extractor.uncompress(Extractor.java:45)
	at <js> Extractor.extract(utils.functions.filesystem.extract:142-144:3469-3595)
	at <js> :anonymous(engines.wine.engine.implementation:170-174:6848-7053)
	at <js> _installRuntime(engines.wine.engine.implementation:158-177:6327-7088)
	at <js> install(engines.wine.engine.implementation:36:1638-1675)
	at <js> run(engines.wine.engine.implementation:297:12220-12253)
	at org.graalvm.sdk/org.graalvm.polyglot.Value.execute(Value.java:338)
	at org.phoenicis.engines.Engine$$JSJavaAdapter.run(Unknown Source)
	at <js> Wine.run(engines.wine.engine.object:178:5495-5585)
	at <js> this.open(engines.wine.plugins.regedit:12:245-309)
	at <js> Wine.luna(engines.wine.verbs.luna:29:956-983)
	at <js> OriginScript.go(engines.wine.quick_script.origin_script:44-47:1242-1391)
	at <js> run(Unnamed:16-49:581-2522)
	at org.graalvm.sdk/org.graalvm.polyglot.Value.executeVoid(Value.java:358)
	at org.phoenicis.scripts.Installer$$JSJavaAdapter.run(Unknown Source)
	at <js> :program(Unnamed:56:2648-2668)
	at org.graalvm.sdk/org.graalvm.polyglot.Context.eval(Context.java:367)
	at org.phoenicis.scripts.engine.implementation.PolyglotScriptEngine.eval(PolyglotScriptEngine.java:63)
	at org.phoenicis.scripts.interpreter.PhoenicisScriptInterpreter.runScript(PhoenicisScriptInterpreter.java:36)
	at org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:37)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by host exception: org.phoenicis.tools.archive.ArchiveException: Unable to extract the file
```

**Comment by Kreyren** [comment]
> I get this after Subnautica Below Zero script is done with the preInstall step.
> 
> ```
> java.lang.IllegalAccessException: access to public member failed: sun.awt.SunToolkit.getScreenSize[Ljava.lang.Object;@77968cff/invokeVirtual, from public Lookup
> 	at org.graalvm.truffle/com.oracle.truffle.polyglot.HostMethodDesc$SingleMethod$MethodMHImpl.makeMethodHandle(HostMethodDesc.java:318)
> 	at org.graalvm.truffle/com.oracle.truffle.polyglot.HostMethodDesc$SingleMethod$MHBase.invoke(HostMethodDesc.java:258)
> 	at org.graalvm.truffle/com.oracle.truffle.polyglot.HostExecuteNode$1.executeImpl(HostExecuteNode.java:776)
> 	at org.graalvm.truffle/com.oracle.truffle.polyglot.GuestToHostRootNode.execute(GuestToHostRootNode.java:87)
> 	at <js> :anonymous(Unnamed:28:1195-1246)
> 	at <js> SteamScript.go(engines.wine.quick_script.steam_script:152:7857-7891)
> 	at <js> run(Unnamed:14-32:502-1389)
> 	at org.graalvm.sdk/org.graalvm.polyglot.Value.executeVoid(Value.java:358)
> 	at org.phoenicis.scripts.Installer$$JSJavaAdapter.run(Unknown Source)
> 	at <js> :program(Unnamed:39:1515-1535)
> 	at org.graalvm.sdk/org.graalvm.polyglot.Context.eval(Context.java:367)
> 	at org.phoenicis.scripts.engine.implementation.PolyglotScriptEngine.eval(PolyglotScriptEngine.java:63)
> 	at org.phoenicis.scripts.interpreter.PhoenicisScriptInterpreter.runScript(PhoenicisScriptInterpreter.java:36)
> 	at org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:37)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
> 	at java.base/java.lang.Thread.run(Thread.java:834)
> Caused by host exception: java.lang.IllegalStateException: java.lang.IllegalAccessException: access to public member failed: sun.awt.SunToolkit.getScreenSize[Ljava.lang.Object;@77968cff/invokeVirtual, from public Lookup
> ```
> 
> Looks like the method we used to get the screen resolution for virtual desktop is obsolete.

Seems irelevant to this issue

**Comment by plata** [comment]
> Seems irelevant to this issue

I agree. The described issue should be very specific for only a few scripts.

**Comment by Kreyren** [comment]
Referencing https://github.com/PhoenicisOrg/scripts/pull/915 which is blocked by this issue.

**Comment by Kreyren** [comment]
Referencing **ALL STEAM SCRIPTS** which are affected by this issue..

**Comment by plata** [comment]
@Zemogiter @ImperatorS79 can you confirm this issue?

**Comment by Zemogiter** [comment]
I don't have hitman absolution on steam but I just tried to reproduce the issue using Space Colony. It installs just fine.

**Comment by Kreyren** [comment]
> I don't have hitman absolution on steam but I just tried to reproduce the issue using Space Colony. It installs just fine.

Can you provide a video of the installation? Tested it on multiple systems and all results in this fatal..

OS used?

**Comment by Zemogiter** [comment]
Ubuntu 19.04 x64 5.0.0-20-generic
https://www.mediafire.com/file/4d8i3bv5fqqqdbx/2019-06-30_10-43-51.flv/file

**Comment by plata** [comment]
I was somewhat expecting this. @Kreyren it seems that the issue is related to your specific setup and not Phoenicis in general.

**Comment by Kreyren** [comment]
@Zemogiter Phoenicis version?

**Comment by Kreyren** [comment]
Investigating.. I had this issue on debian buster and gentoo where each is using different master of phoenicis at the time.

**Comment by Zemogiter** [comment]
The latest commit for both repos. Plus this PR https://github.com/PhoenicisOrg/phoenicis/pull/2024

**Comment by Kreyren** [comment]
latest commit beeing https://github.com/PhoenicisOrg/scripts/commit/6bc41e66341e116e7e4544ad75a83b8434b27e17?

**Comment by Zemogiter** [comment]
That is correct.

**Comment by ImperatorS79** [comment]
Please notice there is an issue with 32 bits runtime that @qparis should fix, which might prevent to run steam (which is a 32 bits apps).

**Comment by qparis** [comment]
What do I have to do?

**Comment by ImperatorS79** [comment]
I mentioned you in #1045.

**Comment by Kreyren** [comment]
Tried to test it today and it results in `Cannot run program "cabextract" (in directory "/home/kreyren/.Phoenicis/containers/wineprefix/Mafia II/drive_c/windows/Fonts"): error=2, No such file or directory`

on https://github.com/PhoenicisOrg/phoenicis/commit/d3acf20270c4867b507f23ec4553dde130f21859

LOG: https://gist.githubusercontent.com/Kreyren/279477661de5113500f84d55f52e7e60/raw/300afeb1b5eec666f896fa954d78af2751f7b85d/dsgasdg

![image](https://user-images.githubusercontent.com/11302521/60450403-18ca9580-9c2a-11e9-9790-ba7296bfd4cf.png)

---

Probably new issue, creating new issue on demand.

**Comment by plata** [comment]
Do you have `cabextract` installed?

**Comment by Kreyren** [comment]
@plata installed `cabextract` which seems to fix the issue.. 

Recommends implementing a pop-up window for this error with more helpful info.

**Comment by Kreyren** [comment]
While installing Mafia II through phoenicis this window appears: 
![image](https://user-images.githubusercontent.com/11302521/60462990-0a40a600-9c4b-11e9-8656-a58d194a8310.png)

**Comment by plata** [comment]
I'm not that familiar with the runtime. @ImperatorS79 can you tell what happened? It seems like fonts are extracted but that should not be possible if it's still loading the runtime json (at least if the fonts are coming from the runtime).

**Comment by ImperatorS79** [comment]
There is no problem, fonts are extracted at the end of the log file, after runtime and wine installation. However, as I stated somewhere earlier, it seems that the "Downoad runtime json..." stays between the messages as the default displayed one.

**Comment by plata** [comment]
Can we fix that issue to avoid misunderstandings like here?

**Comment by ImperatorS79** [comment]
I have not a single idea from where it comes.

**Comment by plata** [comment]
Ok. Then please open a new issue to track this.

**Comment by Kreyren** [comment]
i'm even more confused..

**Comment by plata** [comment]
It's not stuck in this step. The wizard shows something wrong.

**Comment by Kreyren** [comment]
Like it doesn't even open steam and based on output it's not doing anything.. i let it run for like 4 hours before with no result..

**Comment by plata** [comment]
Can you install other apps? Like Notepad++.

**Comment by Kreyren** [comment]
![image](https://user-images.githubusercontent.com/11302521/60534446-1f2c4080-9d02-11e9-9bb8-11dcd952a1a3.png)

Notepad++ works no issue

LOG : pending

**Comment by ImperatorS79** [comment]
The problem might come from the missing libs in the runtime @plata. Without libfreetype, steam will not start.

**Comment by plata** [comment]
In this case we should have some output in `wine.log`.

**Comment by Kreyren** [comment]
> In this case we should have some output in `wine.log`.

invoked manually from phoenicis-wine https://gist.githubusercontent.com/Kreyren/b94b043b5c2a91dfdcae0330a1b8c412/raw/1039b469b8b41047c5eb6159ba761937d8882744/dsgadgs

Phoenicis log: https://gist.githubusercontent.com/Kreyren/0d5452e2c74c14806aea5f3bc1196a2e/raw/7ba87db5bbef06563812554ede7488cef31767fa/dsgsgsfdg

**Comment by plata** [comment]
It complains about pulse.

**Comment by Kreyren** [comment]
> It complains about pulse.

```console
kreyren@rogisek:~$ pulseaudio --version
pulseaudio 12.2
```

**Comment by plata** [comment]
I should have said: libpulse
```
003f:err:module:load_builtin_dll failed to load .so lib for builtin L"winepulse.drv": libpulsecommon-10.0.so: cannot open shared object file: No such file or directory
```

**Comment by plata** [comment]
Closing because the issue doesn't seem to be caused by the script but rather by the system setup.
